corehq/apps/userreports/views.py

```diff
@@ -124,73 +133,108 @@ def create_report(request, domain):
     return _edit_report_shared(request, domain, ReportConfiguration(domain=domain))
 
 
-class ReportBuilderView(TemplateView):
+class ReportBuilderView(BaseDomainView):
 
     @cls_to_view_login_and_domain
-    @upgrade_knockout_js
+    @use_bootstrap3
+    @use_knockout_js
+    @use_select2
+    @use_daterangepicker
     @method_decorator(toggles.REPORT_BUILDER.required_decorator())
     @method_decorator(requires_privilege_raise404(privileges.REPORT_BUILDER))
-    def dispatch(self, request, domain, **kwargs):
-        self.domain = domain
-        return super(ReportBuilderView, self).dispatch(request, domain, **kwargs)
+    def dispatch(self, request, *args, **kwargs):
+        return super(ReportBuilderView, self).dispatch(request, *args, **kwargs)
 
+    @property
+    def section_name(self):
+        return _("Report Builder")
 
-class ReportTypeTileConfiguration(TileConfiguration):
-    def __init__(self, *args, **kwargs):
-        self.analytics_label = kwargs.pop('analytics_label', None)
-        super(ReportTypeTileConfiguration, self).__init__(*args, **kwargs)
+    @property
+    def section_url(self):
+        return reverse(ReportBuilderTypeSelect.urlname, args=[self.domain])
 
 
-class ReportBuilderTypeSelect(ReportBuilderView):
+class ReportBuilderTypeSelect(JSONResponseMixin, ReportBuilderView):
     template_name = "userreports/builder_report_type_select.html"
+    urlname = 'report_builder_select_type'
+    page_title = ugettext_noop('Select Report Type')
```

Comment:
Alright looks like `page_title` does not get translated elsewhere: https://github.com/dimagi/commcare-hq/blob/f5a98d8efe742b22ec62ae873574f9ef33096937/corehq/apps/hqwebapp/views.py#L649-649, so I'll change this to `ugettext_lazy()`.

`report_title`, which is the other place `ugettext_noop()` does get translated later: https://github.com/dimagi/commcare-hq/blob/f5a98d8efe742b22ec62ae873574f9ef33096937/corehq/apps/userreports/views.py#L341-341